adam/learner/__init__.py

```diff
@@ -322,10 +324,26 @@ def concepts_to_patterns(self) -> Dict[Concept, PerceptionGraphPattern]:
 
     @abstractmethod
     def propose_updated_hypotheses(
-        self, concept_to_updated_patterns: Dict[Concept, PerceptionGraphPattern]
+        self,
+        concept_to_updated_patterns: Dict[
```

Comment:
Just occurred to me but this should probably change now that the values are themselves mappings:

```suggestion
        concept_to_hypothesis_updates: Dict[
```